### DISCUSSION TITLE
Prefetching issues with middleware rewriting URL's with base path

### DISCUSSION THREAD

**Discussion by austintreneff** [wpengine/faustjs#1427] [Q&A]
Looking at the router code here: https://github.com/vercel/next.js/blob/6fd866bfd7d4377f5574aa1710eb89343ffad8bb/packages/next/src/shared/lib/router/router.ts#L2445 I am noticing some differences between my local next start versus what is happening when deployed to Vercel. 

Background:
* I am running into some basePath URL rewrite issues that have led me to debugging this area of the router code.
* Locally (via `next start`), `data.json` has data (some local props)
* Once deployed (to Vercel), `data.json` is null
* This causes the `getDataHref` condition to run which is where things go wrong (my base path is included twice in the new href)

My question:
Why is this code checking for `data.json` to be truthy before setting the value to `data.dataHref`
`data.dataHref` can exist without `data.json` existing

Should this [code](https://github.com/vercel/next.js/blob/6fd866bfd7d4377f5574aa1710eb89343ffad8bb/packages/next/src/shared/lib/router/router.ts#L2445) instead look something like:
```
dataHref: data?.dataHref ??
    this.pageLoader.getDataHref({
      href: url,
      asPath: resolvedAs,
      locale: locale,
    }),
```

I am not sure why we would be checking for `data.json` when setting the `dataHref` variables instead of just checking for the `data.dataHref` and setting it if it exist, else fetch it.

**Comment by theodesp** [comment]
Hey @austintreneff maybe you could redirect this question to Vercel Devs. I don't think we will be able to shine much light here since this is next.js internals.

**Reply by austintreneff** [reply]
My apologies, I opened this issue in the wrong tab  🤦 This was meant to be a discussion in the Next.js repository.

Thanks for the quick response!
